emukit/bayesian_optimization/local_penalization_calculator.py

```diff
@@ -51,21 +57,29 @@ def compute_next_points(self, loop_state: LoopState, context: dict=None) -> np.n
         self.acquisition.update_parameters()
 
         # Initialize local penalization acquisition
-        local_penalization_acquisition = LocalPenalization(self.model)
+        local_penalization_acquisition = IntegratedHyperParameterAcquisition(self.model, LocalPenalization)
```

Comment:
That's a behavior change. How impactful is it? Is it worth thinking about preserving old behavior by default, and adding integration over hyperparametrs as an option? I don't have a strong opinion at the moment, just want to hear some thoughts. Also casting in @javiergonzalezh for discussion

Reply:
That makes sense. I think that the ideal here is to automatically detect if the model has hyper-priors and to enable the new behaviour (Integrated.. etc) when that is the case.

Reply:
What's the right way to detect if the model has hyper-priors? Is it to check it's an instance of  IPriorHyperparameters?

Reply:
Yes, that's the way we normally did it so far: `isinstance(model, IPriorHyperparameters)`

Reply:
Thanks - done.